nomos-services/api/src/http/da.rs

```diff
@@ -269,3 +280,83 @@ where
 
     receiver.await?
 }
+
+pub async fn block_peer<B>(handle: &OverwatchHandle, peer_id: PeerId) -> Result<bool, DynError>
+where
+    B: NetworkBackend + 'static + Send,
+    B::Message: PeerMessagesBuilder,
+{
+    let relay = handle.relay::<NetworkService<B>>().connect().await?;
+    let (sender, receiver) = oneshot::channel();
+    let message = B::Message::create_block_message(peer_id, sender);
+    relay
+        .send(DaNetworkMsg::Process(message))
+        .await
+        .map_err(|(e, _)| e)?;
+    Ok(receiver.await?)
+}
+
+pub async fn unblock_peer<B>(handle: &OverwatchHandle, peer_id: PeerId) -> Result<bool, DynError>
+where
+    B: NetworkBackend + 'static + Send,
+    B::Message: PeerMessagesBuilder,
+{
+    let relay = handle.relay::<NetworkService<B>>().connect().await?;
+    let (sender, receiver) = oneshot::channel();
+    let message = B::Message::create_unblock_message(peer_id, sender);
+    relay
+        .send(DaNetworkMsg::Process(message))
+        .await
+        .map_err(|(e, _)| e)?;
+    Ok(receiver.await?)
+}
+
+pub async fn blacklisted_peers<B>(handle: &OverwatchHandle) -> Result<Vec<PeerId>, DynError>
+where
+    B: NetworkBackend + 'static + Send,
+    B::Message: PeerMessagesBuilder,
+{
+    let relay = handle.relay::<NetworkService<B>>().connect().await?;
+    let (sender, receiver) = oneshot::channel();
+    let message = B::Message::create_blacklisted_message(sender);
+    relay
+        .send(DaNetworkMsg::Process(message))
+        .await
+        .map_err(|(e, _)| e)?;
+    Ok(receiver.await?)
+}
+
+// Builder for generating messages for peers (validator and executor).
+pub trait PeerMessagesBuilder {
+    fn create_block_message(peer_id: PeerId, sender: oneshot::Sender<bool>) -> Self;
+    fn create_unblock_message(peer_id: PeerId, sender: oneshot::Sender<bool>) -> Self;
+    fn create_blacklisted_message(sender: oneshot::Sender<Vec<PeerId>>) -> Self;
+}
```

Comment:
Was the long issue here? Somehow I cannot see the messages anymore, at least during this review. Anyway, maybe the confusion comes from a trait returning itself? In traits the `Self` type refers to the type implementing it not the trait itself. So I think this is fine.

Reply:
Yes, this is where the issue is. @ntn-x2 was pointing out that "Builder" was returning `Self` and not `Message`. I ended up renaming it to factory in the end, but actually good observation, Self is the actual type. Thanks for the input, and thank you @ntn-x2 for the correction as well.